Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Operator/FleetManagement.hs

```diff
@@ -135,50 +142,116 @@ castFleetType = \case
+    SMSHelper.sendSMS merchant.id merchantOpCityId (Sms.SendSMSReq message phoneNumber sender) >>= Sms.checkSmsResult
 
-  -- TODO send otp or joining message sms
-  pure Kernel.Types.APISuccess.Success
+  let key = makeFleetLinkOtpKey phoneNumber
+  void $ Redis.setExp key otpCode 3600
 
-postFleetManagementFleetUnlink ::
+  pure $
+    Common.FleetOwnerSearchRes
+      { fleetOwnerId = cast fleetOwner.id,
+        name = fleetOwner.firstName <> " " <> fromMaybe "" fleetOwner.lastName
+      }
+
+postFleetManagementFleetLinkVerifyOtp ::
   Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant ->
   Kernel.Types.Beckn.Context.City ->
   Kernel.Prelude.Text ->
-  Kernel.Prelude.Text ->
+  Common.FleetOwnerVerifyOtpReq ->
   Environment.Flow Kernel.Types.APISuccess.APISuccess
-postFleetManagementFleetUnlink _merchantShortId _opCity fleetOwnerId requestorId = do
-  operator <- B.runInReplica $ QP.findById (Id requestorId :: Id DP.Person) >>= fromMaybeM (PersonNotFound requestorId)
-  unless (operator.role == DP.OPERATOR) $
-    throwError AccessDenied
+postFleetManagementFleetLinkVerifyOtp merchantShortId opCity requestorId req = do
+  operator <- checkOperator requestorId
 
-  fleetOwner <- B.runInReplica $ QP.findById (Id fleetOwnerId :: Id DP.Person) >>= fromMaybeM (FleetOwnerNotFound fleetOwnerId)
+  fleetOwner <- B.runInReplica $ QP.findById (cast req.fleetOwnerId) >>= fromMaybeM (FleetOwnerNotFound (getId req.fleetOwnerId))
   unless (fleetOwner.role == DP.FLEET_OWNER) $
     throwError (InvalidRequest "Invalid fleet owner")
 
-  QFOA.endFleetOperatorAssociation fleetOwner.id operator.id
+  checkAssocOperator <- B.runInReplica $ QFOA.findByFleetOwnerIdAndOperatorId fleetOwner.id operator.id True
+  when (isJust checkAssocOperator) $ throwError (InvalidRequest "Fleet already associated with operator")
 
-  -- TODO send sms about unlink
+  decryptedMobileNumber <- mapM decrypt fleetOwner.mobileNumber >>= fromMaybeM (InvalidRequest $ "Person does not have a mobile number " <> getId fleetOwner.id)
+  let key = makeFleetLinkOtpKey (fromMaybe "+91" fleetOwner.mobileCountryCode <> decryptedMobileNumber)
+
+  storedOtp <- Redis.get key >>= fromMaybeM (InvalidRequest "OTP expired or not found")
+  unless (storedOtp == req.otp) $ throwError (InvalidRequest "Invalid OTP")
+
+  merchant <- findMerchantByShortId merchantShortId
+  merchantOpCityId <- CQMOC.getMerchantOpCityId Nothing merchant (Just opCity)
+  SA.endFleetAssociationsIfAllowed merchant merchantOpCityId fleetOwner
+
+  fleetOperatorAssociation <- SA.makeFleetOperatorAssociation merchant.id merchantOpCityId (getId fleetOwner.id) operator.id.getId (DomainRC.convertTextToUTC (Just "2099-12-12"))
+  QFOA.create fleetOperatorAssociation
+
+  let phoneNumber = fromMaybe "+91" fleetOwner.mobileCountryCode <> decryptedMobileNumber
+      message = "Operator " <> operator.firstName <> maybe "" (" " <>) operator.lastName <> " successfully linked to your fleet"
```

Comment:
Can we make this message configurable using `MessageBuilder` and db migration, similar as `BuildOperatorJoiningMessageReq`?